File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/prism/PrismFightStateUpdateMessage.scala





















// Generated on 05/03/2013 10:34:59
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PrismFightStateUpdateMessage extends MessageDeserializer {
  val messageId = 6040

  protected trait Message extends BaseMessage with Serializable {
    val messageId = PrismFightStateUpdateMessage.messageId

    val state: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(state)
    }

  }

  protected def create(state: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val state = buf.readByte()

    create(state)
  }

}

object PrismFightStateUpdateMessage extends PrismFightStateUpdateMessage {
  case class MessageImpl(state: Byte) extends Message
  type Target = MessageImpl

  override def create(state: Byte) = MessageImpl(state)

  def apply(state: Byte) = create(state)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PrismFightStateUpdateMessage.java


// Generated on 05/08/2013 19:37:57
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PrismFightStateUpdateMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6040;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte state;
    
    public PrismFightStateUpdateMessage() { }
    
    public PrismFightStateUpdateMessage(byte state) {
        this.state = state;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(state);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        state = buf.readByte();
        if (state < 0)
            throw new RuntimeException("Forbidden value on state = " + state + ", it doesn't respect the following condition : state < 0");
    }
    
}
